### ISSUE TITLE
对CPU设备的支持性问题以及requirements.txt的修改

### ISSUE THREAD

**Issue by Moeary** [stepfun-ai/Step-Audio-EditX#19]
#### 问题1 没有cuda的torch
我根据官方给的例子在本地搭建时,利用anaconda创建了虚拟环境后,使用
```shell
pip install -r requirements.txt
```
命令,下载好所有的依赖,然后启动
```shell
python app.py --model-path ./models --model-source local  
```
时发现提示报错
```shell
2025-11-13 23:27:32,005 - model_loader - INFO - Successfully loaded FunASR model from local
2025-11-13 23:27:32,476 - __main__ - INFO - ✓ StepAudioTokenizer loaded successfully
2025-11-13 23:27:32,477 - tts - INFO - 🔧 StepAudioTTS loading configuration:
2025-11-13 23:27:32,477 - tts - INFO -    - model_source: local
2025-11-13 23:27:32,477 - tts - INFO -    - model_path: models\Step-Audio-EditX
2025-11-13 23:27:32,477 - tts - INFO -    - tts_model_id: models\Step-Audio-EditX
2025-11-13 23:27:32,477 - model_loader - INFO - Loading Transformers model from local: models\Step-Audio-EditX
2025-11-13 23:27:33,015 - model_loader - ERROR - Failed to load model from local: Torch not compiled with CUDA enabled
2025-11-13 23:27:33,015 - tts - ERROR - ❌ Failed to load model: Torch not compiled with CUDA enabled
2025-11-13 23:27:33,015 - __main__ - ERROR - ❌ Error loading models: Torch not compiled with CUDA enabled
2025-11-13 23:27:33,015 - __main__ - ERROR - Please check your model paths and source configuration.
```shell
2025-11-13 23:10:08,956 - __main__.EditxTab - INFO - Starting voice cloning process
2025-11-13 23:10:08,956 - __main__.EditxTab - ERROR - [Error] CLONE button must use clone task.
2025-11-13 23:10:15,173 - __main__.EditxTab - INFO - Starting voice cloning process
2025-11-13 23:10:27,433 - tts - ERROR - Clone failed: Torch not compiled with CUDA enabled
2025-11-13 23:10:27,434 - __main__.EditxTab - ERROR - [Error] Clone failed: Torch not compiled with CUDA enabled
2025-11-13 23:10:45,148 - __main__.EditxTab - INFO - Starting audio editing process
2025-11-13 23:10:45,305 - tts - ERROR - Edit failed: Torch not compiled with CUDA enabled
2025-11-13 23:10:45,305 - __main__.EditxTab - ERROR - [Error] Edit failed: Torch not compiled with CUDA enabled
```
就去查看了下pip下载的包发现带cuda的torch没下载下来,具体见于(Python包)那一块
然后又跑过去看了下requirements.txt的torch版本是不带cuda的

#### 应该的解决办法
应该考虑在readme里面填写一下引导用户使用其他镜像源直接下载带cuda的torch,放在
```shell
pip install -r requirements.txt
```
的前面以免各种包版本配置的污染

#### 问题2  不使用torch with cuda版本的纯cpu运行不起来
在遇到前面的问题后我尝试使用
```shell
python app.py --model-path models --model-source local --device-map cpu
```
来直接使用cpu跑
gradio的demo网页是能正常显示的,提示有
```shell
2025-11-13 23:08:13,375 - __main__ - INFO - ✓ StepAudioTokenizer loaded successfully
2025-11-13 23:08:13,375 - tts - INFO - 🔧 StepAudioTTS loading configuration:
2025-11-13 23:08:13,375 - tts - INFO -    - model_source: local
2025-11-13 23:08:13,375 - tts - INFO -    - model_path: models\Step-Audio-EditX
2025-11-13 23:08:13,375 - tts - INFO -    - tts_model_id: models\Step-Audio-EditX
2025-11-13 23:08:13,376 - model_loader - INFO - Loading Transformers model from local: models\Step-Audio-EditX
2025-11-13 23:08:14,429 - model_loader - INFO - Successfully loaded model from local
2025-11-13 23:08:14,430 - tts - INFO - ✅ Successfully loaded LLM and tokenizer: models\Step-Audio-EditX
2025-11-13 23:08:15,914 - tts - INFO - 🎤 CosyVoice model loaded successfully
2025-11-13 23:08:15,914 - __main__ - INFO - ✓ StepCommonAudioTTS loaded successfully
* Running on local URL:  http://0.0.0.0:7860
```
然后我进浏览器输入了参考音频和参考音频文本 还有目标文本 无论是使用clone模式后还是edit模式后都会提示没cuda支持就别想跑了 问题出在tts模型上,理应来说所有的模型都应该能让cpu来跑的,但是这个不支持,可能是你们传参的时候忘记传cpu参数进去了?
```shell
2025-11-13 23:10:08,956 - __main__.EditxTab - INFO - Starting voice cloning process
2025-11-13 23:10:08,956 - __main__.EditxTab - ERROR - [Error] CLONE button must use clone task.
2025-11-13 23:10:15,173 - __main__.EditxTab - INFO - Starting voice cloning process
2025-11-13 23:10:27,433 - tts - ERROR - Clone failed: Torch not compiled with CUDA enabled
2025-11-13 23:10:27,434 - __main__.EditxTab - ERROR - [Error] Clone failed: Torch not compiled with CUDA enabled
2025-11-13 23:10:45,148 - __main__.EditxTab - INFO - Starting audio editing process
2025-11-13 23:10:45,305 - tts - ERROR - Edit failed: Torch not compiled with CUDA enabled
2025-11-13 23:10:45,305 - __main__.EditxTab - ERROR - [Error] Edit failed: Torch not compiled with CUDA enabled
```

#### 应该的解决办法
修改源代码 使其更好的支持不带cuda版本的torch

#### 问题3 cosyvoice相关
问一下你们这个项目和cosyvoice有什么关系,并无恶意,因为我看到两个都是apache2协议的 不存在任何版权纠纷 只是因为我偶然瞟了一眼看到你们Step-Audio-EditX的模型文件夹下面和一个stepvocoder文件夹下面都有cosyvoice来着,这个就是最终的tts模型吗

### 环境
Windows10 2022H2
Anaconda环境+Python3.10

### Python包
```shell
accelerate             1.10.0
aiofiles               24.1.0
aiohappyeyeballs       2.6.1
aiohttp                3.13.2
aiosignal              1.4.0
aliyun-python-sdk-core 2.16.0
aliyun-python-sdk-kms  2.16.5
annotated-doc          0.0.4
annotated-types        0.7.0
antlr4-python3-runtime 4.9.3
anyio                  4.11.0
async-timeout          5.0.1
attrs                  25.4.0
audioread              3.1.0
brotli                 1.2.0
certifi                2025.11.12
cffi                   2.0.0
charset-normalizer     3.4.4
click                  8.3.0
colorama               0.4.6
coloredlogs            15.0.1
compressed-tensors     0.11.0
conformer              0.3.2
contourpy              1.3.2
crcmod                 1.7
cryptography           46.0.3
cycler                 0.12.1
datasets               4.0.0
decorator              5.2.1
diffusers              0.35.2
dill                   0.3.8
editdistance           0.8.1
einops                 0.8.1
exceptiongroup         1.3.0
fastapi                0.121.1
ffmpy                  1.0.0
filelock               3.20.0
flatbuffers            25.9.23
fonttools              4.60.1
frozendict             2.4.7
frozenlist             1.8.0
fsspec                 2025.3.0
funasr                 1.2.7
gradio                 5.49.1
gradio_client          1.13.3
groovy                 0.1.2
h11                    0.16.0
httpcore               1.0.9
httpx                  0.28.1
huggingface-hub        0.36.0
humanfriendly          10.0
hydra-core             1.3.2
HyperPyYAML            1.2.2
idna                   3.11
importlib_metadata     8.7.0
jaconv                 0.4.0
jamo                   0.4.1
jieba                  0.42.1
Jinja2                 3.1.6
jmespath               0.10.0
joblib                 1.5.2
kaldiio                2.18.1
kiwisolver             1.4.9
lazy_loader            0.4
librosa                0.10.2.post1
llmcompressor          0.7.0
llvmlite               0.45.1
loguru                 0.7.3
markdown-it-py         4.0.0
MarkupSafe             3.0.3
matplotlib             3.10.7
mdurl                  0.1.2
modelscope             1.31.0
more-itertools         10.8.0
mpmath                 1.3.0
msgpack                1.1.2
multidict              6.7.0
multiprocess           0.70.16
networkx               3.4.2
numba                  0.62.1
numpy                  2.2.6
nvidia-cuda-nvrtc-cu12 12.8.93
nvidia-ml-py           12.575.51
omegaconf              2.3.0
onnxruntime            1.23.2
onnxruntime-gpu        1.17.0
openai-whisper         20240930
orjson                 3.11.4
oss2                   2.19.1
packaging              25.0
pandas                 2.3.3
pillow                 10.4.0
pip                    25.3
platformdirs           4.5.0
pooch                  1.8.2
propcache              0.4.1
protobuf               5.29.3
psutil                 5.9.8
pyarrow                22.0.0
pycparser              2.23
pycryptodome           3.23.0
pydantic               2.11.10
pydantic_core          2.33.2
pydub                  0.25.1
Pygments               2.19.2
pynndescent            0.5.13
pynvml                 12.0.0
pyparsing              3.2.5
pyreadline3            3.5.4
python-dateutil        2.9.0.post0
python-multipart       0.0.20
pytorch-wpe            0.0.1
pytz                   2025.2
PyYAML                 6.0.2
regex                  2025.11.3
requests               2.32.5
rich                   14.2.0
ruamel.yaml            0.18.16
ruamel.yaml.clib       0.2.14
ruff                   0.14.4
safehttpx              0.1.7
safetensors            0.6.2
scikit-learn           1.7.2
scipy                  1.15.3
semantic-version       2.10.0
sentencepiece          0.2.1
setuptools             80.9.0
shellingham            1.5.4
six                    1.16.0
sniffio                1.3.1
soundfile              0.13.1
sox                    1.5.0
soxr                   1.0.0
spaces                 0.42.1
starlette              0.49.3
sympy                  1.14.0
tensorboardX           2.6.4
threadpoolctl          3.6.0
tiktoken               0.12.0
tokenizers             0.21.4
tomlkit                0.13.3
torch                  2.8.0
torch-complex          0.4.4
torchaudio             2.8.0
torchvision            0.23.0
tqdm                   4.67.1
transformers           4.53.3
typer                  0.20.0
typing_extensions      4.15.0
typing-inspection      0.4.2
tzdata                 2025.2
umap-learn             0.5.9.post2
urllib3                2.5.0
uvicorn                0.38.0
websockets             15.0.1
wheel                  0.45.1
win32_setctime         1.2.0
xxhash                 3.6.0
yarl                   1.22.0
zipp                   3.23.0
```
